test/format/t.c

```diff
@@ -358,7 +353,8 @@ format_die(void)
     /*
      * Single-thread error handling, our caller exits after calling us (we never release the lock).
      */
-    (void)pthread_rwlock_wrlock(&g.death_lock);
+    if (LOCK_INITIALIZED(&g.death_lock))
+        (void)pthread_rwlock_wrlock(&g.death_lock.l.pthread);
```

Comment:
I'm a little hesitant to use a WT read/write lock for the death-lock, given that we're here because the library has gone south somehow. I can't think of any reason the locks wouldn't keep working, but figured I'd mention it in case you have thought of something I haven't.

Reply:
I was simply treating all the locks the same.  But you raise a good point.  I've changed the death-lock back to always use a pthread rwlock.  It's not as if performance matters for that use case :-)